src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1170,7 +1170,7 @@ protected virtual void ReportUnassigned(Symbol symbol, SyntaxNode node, int slot
 
             if (skipIfUseBeforeDeclaration &&
                 symbol.Kind == SymbolKind.Local &&
-                (symbol.Locations.Length == 0 || node.Span.End < symbol.GetFirstLocationOrNone().SourceSpan.Start))
+                (symbol.TryGetFirstLocation() is var location && (location is null || node.Span.End < location.SourceSpan.Start)))
```

Comment:
I'm not a fan of treating a canonical `Try*` failure path as a sneaky true in this expression. Does this work?

```suggestion
                (symbol.TryGetFirstLocation() is not { } location || (node.Span.End < location.SourceSpan.Start)))
```